students/ScotchWSplenda/lesson10/assignment/database10.py

```diff
@@ -0,0 +1,197 @@
+'''
+table = collection.
+row = document.
+column = field.
+Database = database
+index = index
+'''
+import csv
+# import logging
+from pymongo import MongoClient
+from pprint import pprint
+import logging
+import time
+import datetime
+
+
+class MongoDBConnection(object):
+    """MongoDB Connection"""
+    def __init__(self, host='127.0.0.1', port=27017):
+        """ be sure to use the ip address not name for local windows"""
+        self.host = host
+        self.port = port
+        self.connection = None
+
+    def __enter__(self):
+        self.connection = MongoClient(self.host, self.port)
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.connection.close()
+
+
+def timer_decoration(func):
+    def wrapper(*args, **kwargs):
+        # define the format and tell the logging module about your format
+        formatter = logging.Formatter('%(asctime)s %(filename)s:%(lineno)-3d %(levelname)s %(message)s')
+        # tell the logging module HOW to save in its file (saves in current dir)
+        FH = logging.FileHandler(datetime.datetime.now().strftime('%Y-%m-%d')+'.log')
+        FH.setFormatter(formatter)
+        # activate logger
+        LOGGER = logging.getLogger()
+        LOGGER.addHandler(FH)
```

Comment:
Your logger continues to exist beyond the decorator, so every handler you add piles up with the previous handlers, the duplicated (and triplicated) output you see is a consequence of all these loggers running. A lazy solution is to have your decorator clean all previous handlers, just add this in the previous line:

`        LOGGER.handlers = []`

Reply:
After doing that, the duplicated lines are gone:

2020-08-18 19:48:50,706 database10.py:51  INFO import_data took 0.17222189903259277 seconds.
2020-08-18 19:48:50,728 database10.py:51  INFO show_available_products took 0.02124500274658203 seconds.
2020-08-18 19:48:50,903 database10.py:51  INFO import_data took 0.17414093017578125 seconds.